### PR TITLE
[macos-claude] Fix: recover login-shell PATH on macOS so claude/copilot can launch

### PR DESCRIPTION
**Pull request by mcaden**
## Summary

- macOS `launchd` starts `.app` bundles with a minimal `PATH` (`/usr/bin:/bin:/usr/sbin:/sbin` + `/etc/paths.d`) and never sources the user's shell rc files. The PTY pool spawns each session as `$SHELL -c <cmd>` (non-interactive, non-login), so `~/.zshrc` doesn't run before the spawn either. User-installed CLIs (`~/.local/bin`, `~/.npm-global/bin`, Homebrew, etc.) are invisible — the user sees `claude: command not found` from a Finder-started build even though Terminal.app finds it fine.
- Adds a new module `src-tauri/src/login_path.rs` that runs once at boot on macOS: spawns `<$SHELL> -ilc 'printf MARKER; echo $PATH'`, splits on a sentinel marker (so chatty rc-file output can't poison the result), and applies the recovered `PATH` via `std::env::set_var`. Every subsequent PTY child inherits it.
- Failure modes (timeout, missing `$SHELL`, parse error, non-zero exit) are logged at `warn!` and leave `PATH` untouched — boot never aborts. macOS only; `apply_login_path_macos()` is a `cfg`-gated no-op everywhere else.

## Why this approach

This is the same fix pattern as VS Code's `resolveShellEnv` and `npm:fix-path` — query the user's *interactive login* shell once, lift the resolved `PATH` into the host process so all children inherit. `-ilc` is required because macOS users put `PATH` additions in `~/.zshrc` (interactive) rather than `~/.zprofile` (login-only). Scope is intentionally tight: only `PATH` is lifted; the rest of the env is left alone to avoid clobbering Tauri-provided variables.

Hooked into `lib::run()`'s `.setup` closure between `init_tracing` and `boot::parse_cli_args` so the corrected `PATH` is in place before:
- `config_store::seed_default_custom_processes` probes for `code` (VS Code launcher default-enable),
- the PTY pool spawns any session,
- `app_launcher::which_in_path` preflight runs.

Linux is out of scope — GUI launchers there (gdm/sddm/.desktop) inherit the user's session PATH and we have no reports of the symptom. The `cfg` guard is easy to relax later if needed.

## Test plan

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --features test-helpers -- -D warnings`
- [x] `cargo test --workspace --features test-helpers` (564 lib tests + all integration suites pass; 9 new `login_path` unit tests cover parser edge cases: marker found, marker missing, CRLF, blank lines, rc-file noise stripped, empty PATH, query error propagation)
- [x] `pnpm run lint`
- [x] `pnpm test --run` (678 tests pass)
- [x] `pnpm run build`
- [ ] **E2E on macOS (cannot run on Windows host)** — install this build, launch from Finder/Dock (not Terminal — Terminal would mask the bug by inheriting its own PATH), open a Claude session, expect claude's TUI banner instead of `command not found`. If still failing, the resolved PATH is logged at `info` in `~/Library/Logs/com.arborist.app/arborist.log` (or wherever `app_log_dir` resolves to); the relevant log line is either `applying login-shell PATH` or `login-shell PATH query failed`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src-tauri/src/login_path.rs [hunk 1]
```diff
@@ -0,0 +1,227 @@
+//! Boot-time login-shell `PATH` recovery for macOS `.app` launches.
+//!
+//! `launchd` starts `.app` bundles with a minimal `PATH` (`/usr/bin:/bin:/usr/sbin:/sbin` plus `/etc/paths.d`) and does **not** source any user shell
+//! rc files. The PTY pool later spawns each session as `<$SHELL> -c <cmd>` (non-interactive, non-login), so user-installed CLIs in `~/.local/bin`,
+//! `~/.npm-global/bin`, Homebrew prefixes, etc. are invisible to the spawned child — `claude: command not found` is the typical user-visible symptom.
+//!
+//! Fix: at boot we ask the user's login shell what its `PATH` is via `<$SHELL> -ilc 'printf MARKER\n%s\n "$PATH"'`, parse the line after the marker,
+//! and apply it to the host process env so every subsequent PTY child inherits the corrected `PATH`. Same pattern as `npm:fix-path` /
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/login_path.rs]: 🤖 AI agent reply (acting for @mcaden):

Done in 42db008e. Module doc now reads `printf '%s\n%s\n' MARKER "$PATH"` to match the actual script. PR description left as-is for now since it's the higher-level summary; happy to update that too if you'd like.

File: src-tauri/src/login_path.rs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+pub fn apply_login_path_macos() {
+    match compute(&RealLoginShellQuery) {
+        Ok(Some(p)) => {
+            tracing::info!(path = %p, "applying login-shell PATH");
+            std::env::set_var("PATH", p);
+        }
+        Ok(None) => tracing::warn!("login-shell PATH was empty; keeping inherited PATH"),
+        Err(e) => tracing::warn!(error = %e, "login-shell PATH query failed; keeping inherited PATH"),
+    }
+}
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/login_path.rs]: 🤖 AI agent reply (acting for @mcaden):

Good catch. Fixed in 42db008e by dropping the dead branch — `compute` now returns `Result<String, _>` directly, and the parser already errors with `NoPathAfterMarker` when no non-empty line follows the marker. I didn't take the alternative interpretation (treat a literally empty `PATH` as a success) because a user whose login shell deliberately exports `PATH=` would still get `command not found` on every CLI; refusing to clobber an inherited `/usr/bin:/bin` PATH in that case seems strictly better than honoring the empty value.

File: src-tauri/src/login_path.rs [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            .map_err(|e| LoginPathError::Spawn(io::Error::other(format!("thread spawn failed: {e}"))))?;
+
+        let output = match rx.recv_timeout(QUERY_TIMEOUT) {
+            Ok(Ok(out)) => out,
+            Ok(Err(e)) => return Err(LoginPathError::Spawn(e)),
+            Err(mpsc::RecvTimeoutError::Timeout) => return Err(LoginPathError::Timeout(QUERY_TIMEOUT)),
+            Err(mpsc::RecvTimeoutError::Disconnected) => {
+                return Err(LoginPathError::Spawn(io::Error::other("shell thread disconnected")));
+            }
+        };
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/login_path.rs]: 🤖 AI agent reply (acting for @mcaden):

Fixed in 42db008e. The shell is now spawned via `Command::spawn()` so we keep the `Child` handle. The exit-status check is a `try_wait`/sleep poll loop with a 50ms cadence; on `Timeout` (or a `try_wait` error) we `child.kill()` + `child.wait()`, which causes the piped stdout/stderr to EOF and lets the two reader threads join cleanly. No orphaned zsh, no leaked threads. Stdout and stderr are drained on dedicated threads (matching what `Command::output()` did internally) so a chatty rc-file `>&2` can't fill the pipe buffer and deadlock the child before its `printf` line.

File: src-tauri/src/lib.rs [hunk 4]
```diff
@@ -145,6 +146,9 @@ pub fn run() {
             let log_guard = init_tracing(Some(&log_dir));
             tracing::info!("Arborist starting up");
 
+            // Recover the user's interactive PATH before anything inherits this process's env (macOS only; no-op elsewhere).
+            login_path::apply_login_path_macos();
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/lib.rs:150]: 🤖 AI agent reply (acting for @mcaden):

Leaving the 5s budget for now. Boot sequencing: while the Tauri `.setup` closure runs, the main window is `visible: false` (lib.rs:363), so the host shows the macOS dock-bounce as its "launching" affordance. Once setup returns, the WebView immediately renders `<BootTreeAnimation />` + `Loading Arborist…` from `App.tsx:92-93` until the bound/unbound state settles. So the user always sees a launching indicator during this wait — never a frozen blank screen.

And the "proper error" path is already user-visible without extra UI: if the timeout fires we keep the inherited minimal PATH, the macOS shell-out for `claude` fails, and the session terminal shows `claude: command not found` — which is the same diagnostic the user would have hit before this fix existed. The `warn!` line in the log gives the developer-facing detail.

Tightening the budget to 1–1.5s would risk false negatives on the very setups this fix exists for (oh-my-zsh + p10k + nvm + pyenv hits 1–2s on slower hardware), and moving to a deferred-apply background thread would mean any PTY spawned before the apply lands still sees the broken PATH — which defeats the whole point. Open to revisiting if telemetry later shows P95 boot wait above a threshold worth fixing.

File: src-tauri/src/login_path.rs [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+#[cfg(target_os = "macos")]
+fn resolve_shell() -> Result<String, LoginPathError> {
+    // launchd sets $SHELL from Directory Services, so this is reliable on macOS. Fallback to /bin/zsh (the default user shell on Catalina+).
+    Ok(std::env::var("SHELL").unwrap_or_else(|_| "/bin/zsh".to_owned()))
+}
+
+#[cfg(not(target_os = "macos"))]
+fn resolve_shell() -> Result<String, LoginPathError> {
+    std::env::var("SHELL").map_err(|_| LoginPathError::NoShell)
+}
```
[the remark this answers is not in the document]
Comment (mcaden) [reply src-tauri/src/login_path.rs:141]: 🤖 AI agent reply (acting for @mcaden):

Leaving as-is. The suggested pre-validation doesn't actually prevent the case raised: a user-set `$SHELL=/usr/bin/python` (or any other interactive REPL) is fully executable, would pass a `which`/`is_file` + exec-bit check, and would still hang on the `-ilc` flag — landing us right back in the `Timeout` path. The only thing pre-validation would buy is converting a `Spawn(ENOENT)` into a custom `NotAShell`/`NoShell` variant for "binary missing entirely", which is already covered by the inherited `io::Error` in the `Spawn` variant.

The current degradation chain — `spawn fail → Spawn(_)`, `hangs → Timeout`, `exits non-zero → NonZeroExit`, `garbage output → MarkerNotFound` — all funnel to the same `warn!` + keep-inherited-PATH outcome, and the user still gets the concrete `command not found` at the session terminal. Happy to revisit if a real `$SHELL` configuration surfaces that this would catch and the current path wouldn't.
